hedera-node/src/test/java/com/hedera/services/store/contracts/HederaStackedWorldStateUpdaterTest.java

```diff
@@ -0,0 +1,92 @@
+package com.hedera.services.store.contracts;
+
+/*
+ * -
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ *
+ */
+
+
+import com.hedera.services.utils.EntityIdUtils;
+import com.hederahashgraph.api.proto.java.ContractID;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.evm.worldstate.AbstractWorldUpdater;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.BDDMockito.given;
+
+@ExtendWith(MockitoExtension.class)
+class HederaStackedWorldStateUpdaterTest {
+
+	@Mock
+	private AbstractWorldUpdater<HederaWorldState, HederaWorldState.WorldStateAccount> updater;
+	@Mock
+	private HederaMutableWorldState worldState;
+
+	private HederaStackedWorldStateUpdater subject;
+
+	@BeforeEach
+	void setUp() {
+		subject = new HederaStackedWorldStateUpdater(updater, worldState);
+	}
+
+	@AfterEach
+	void tearDown() {
+		subject.getSponsorMap().clear();
+	}
+
+	@Test
+	void allocateNewContractAddress() {
+		final var sponsorAddr = Address.wrap(Bytes.wrap(EntityIdUtils.asSolidityAddress(
+				ContractID.newBuilder().setContractNum(1).build())));
+
+		final var sponsorAddr2 = Address.wrap(Bytes.wrap(EntityIdUtils.asSolidityAddress(
+				ContractID.newBuilder().setContractNum(2).build())));
+		given(worldState.newContractAddress(sponsorAddr)).willReturn(sponsorAddr2);
```

Comment:
since sponsorAddr2 is the address that is being sponsored, maybe it would make more sense to rename the var to something like "sponsoredAddr"